psamm/datasource/native.py

```diff
@@ -174,28 +174,38 @@ class NativeModel(object):
     directory subtree that specifies part of the model.
     """
 
-    def __init__(self, path):
+    def __init__(self, model_from):
         """Create a model from the specified model file or directory"""
-
-        if os.path.isfile(path):
-            self._context = FilePathContext(path)
-            with open(self._context.filepath, 'r') as f:
-                self._model = yaml_load(f)
-        else:
-            # Try to open the default file
-            for filename in DEFAULT_MODEL:
-                try:
-                    self._context = FilePathContext(
-                        os.path.join(path, filename))
-                    with open(self._context.filepath, 'r') as f:
-                        self._model = yaml_load(f)
-                        break
-                except:
-                    logger.debug('Failed to load model file', exc_info=True)
+        if type(model_from) is str:
```

Comment:
Checking a type should usually be done with `if isinstance(<var>, <types>)...` instead of `if type(<var>) is <type>...` (https://stackoverflow.com/questions/21894575/isinstancefoo-bar-vs-typefoo-is-bar). This also applies to the other type checks below. Also, checking string types should use `string_types` from the `six` module instead of `str` because of the changes between Python 2 and 3.